spring-cloud-stream/src/main/java/org/springframework/cloud/stream/binder/AbstractBinder.java

```diff
@@ -64,6 +66,12 @@
 
 	private volatile EvaluationContext evaluationContext;
 
+	protected final BinderErrorConfigurer errorConfigurer;
+
+	protected AbstractBinder(BinderErrorConfigurer errorConfigurer) {
+		this.errorConfigurer = errorConfigurer;
```

Comment:
Do you think we can probably make `BinderErrorConfigurer` into setter level injection rather than constructor level injection? 

May be that could help these changes less invasive in terms of creating the Binder instance and also, the implementations wouldn't have to worry about setting this to `null` explicitly.